src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -218,17 +230,15 @@ public void taskCompletedWithProblems(Executor executor, Queue.Task task, long d
         postJobAction(executor);
     }
 
-    private void postJobAction(Executor executor){
+    private void postJobAction(Executor executor) {
         EC2Computer computer = (EC2Computer) executor.getOwner();
         EC2AbstractSlave slaveNode = computer.getNode();
-        int maxTotalUses = slaveNode.maxTotalUses;
-        if (maxTotalUses <= -1) {
-            return;
-        } else if (maxTotalUses <= 1) {
-            computer.setAcceptingTasks(false);
-            slaveNode.terminate();
+        // At this point, if agent is in suspended state and has 1 last executer running, it is safe to terminate.
+	    if (computer.countBusy() <= 1 && !computer.isAcceptingTasks()) {
+            LOGGER.info("Agent " + slaveNode.instanceId + " is terminated due to maxTotalUses (" + slaveNode.maxTotalUses + ")");
+	        slaveNode.terminate();
         } else {
-            slaveNode.maxTotalUses = slaveNode.maxTotalUses - 1;
+            LOGGER.info("Agent " + slaveNode.instanceId + " is still in use by more than one (" + computer.countBusy() + ") executers");
```

Comment:
This will log after _every_ task. May need to guard it better so it only emits when applicable.

Reply:
Something like this?
```if (maxTotalUses <= 1) {
    LOGGER.info("Agent " + slaveNode.instanceId + " is still in use by more than one (" + computer.countBusy() + ") executers");
}```

Reply:
Yeah, I would only want to see this message when an instance is slated to be removed and the `maxTotalUses` feature is in use.